test/org/jetbrains/java/decompiler/CommandLineTest.java

```diff
@@ -48,6 +53,30 @@ public void testJarToDir() {
     assertFilesEqual(fixture.getTestDataDir().resolve("bulk"), fixture.getTempDir().resolve("bulk_out"));
   }
 
+  @Test
+  public void testMaliciousJarToDir() throws IOException {
+    String out = fixture.getTempDir().resolve("bulk_out").toAbsolutePath().toString();
+    String in = fixture.getTestDataDir().resolve("mal.jar").toAbsolutePath().toString();
+    // create malicious jar if doesn't exist already
+    if (!new File(in).exists())
+    {
+      String badFileName = new String("../../../../../../../../../../../tmp/test.txt");
+      try (ZipOutputStream maliciousOut = new ZipOutputStream(new FileOutputStream(fixture.getTestDataDir().resolve("mal.jar").toAbsolutePath().toString())))
+      {
+          ZipEntry entry = new ZipEntry(badFileName);
+          maliciousOut.putNextEntry(entry);
+          maliciousOut.write("womp womp".getBytes());
+          maliciousOut.closeEntry();
+      }
+    }
+
+    ConsoleDecompiler.main(new String[]{in, out});
+
+    TextBuffer.checkLeaks();
+    File f = new File("/tmp/test.txt");
```

Comment:
I would suggest using a more distinct file name... I could just have a `/tmp/test.txt` already on my system

Reply:
JUnit also has a way to generate a test-specific temp file directory, that's probably the way to go